src/shared/kbd-util.c

```diff
@@ -3,14 +3,62 @@
 #include "errno-util.h"
 #include "kbd-util.h"
 #include "log.h"
-#include "nulstr-util.h"
 #include "path-util.h"
 #include "recurse-dir.h"
 #include "set.h"
 #include "string-util.h"
 #include "strv.h"
 #include "utf8.h"
 
+#define KBD_KEYMAP_DIRS                         \
+        "/usr/share/keymaps/",                  \
+        "/usr/share/kbd/keymaps/",              \
+        "/usr/lib/kbd/keymaps/"
+
+int keymap_directories(char ***ret) {
+        _cleanup_strv_free_ char **paths = NULL;
+        const char *e;
+        int r;
+
+        assert(ret);
+
+        e = getenv("SYSTEMD_KEYMAP_DIRECTORIES");
+        if (!e)
+                goto use_default;
+
+        r = strv_split_full(&paths, e, NULL, EXTRACT_UNQUOTE);
+        if (r == 0) /* empty string? */
+                goto use_default;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse $SYSTEMD_KEYMAP_DIRECTORIES environment variable, ignoring: %m");
+                goto use_default;
+        }
+
+        STRV_FOREACH(p, paths) {
+                if (!path_is_absolute(*p)) {
+                        log_debug("Path specified in $SYSTEMD_KEYMAP_DIRECTORIES environment variable is not absolute, ignoring: %m");
+                        goto use_default;
+                }
+                if (!path_is_normalized(*p)) {
+                        log_debug("Path specified in $SYSTEMD_KEYMAP_DIRECTORIES environment variable is not normalized, ignoring: %m");
+                        goto use_default;
+                }
+        }
+
+        *ret = TAKE_PTR(paths);
+        return 0;
```

Comment:
i think it would be nice to have a helper `static int keymap_directories_from_environment()` or so that just does the env var parsing, and returns a regular negative errno error. And then keymap_directories() just wraps that and implements the fallback.

benefit: no `goto use_default` everywhere.

Reply:
hmm, thinking about this: don't we have a generic helper for this already? it sounds like something we should have if we don't. After all acquiring a set of search paths from an env var is a pretty common thing no?

Reply:
ah, yes we do have that:

getenv_path_list()

Might need some tweaks to make it more generic. For example, maybe should return a different error than just EINVAL if no valid paths are specified, so that callers which are fine with it can handle it grcefully.

Reply:
> Might need some tweaks to make it more generic. For example, maybe should return a different error than just EINVAL if no valid paths are specified, so that callers which are fine with it can handle it grcefully.

I agree with that, but I'd like to leave that to another PR.